src/Common/src/System/Net/ContextAwareResult.cs

```diff
@@ -328,17 +363,26 @@ protected override void Cleanup()
         // Returns whether the operation completed sync or not.
         private bool CaptureOrComplete(ref ExecutionContext cachedContext, bool returnContext)
         {
-            GlobalLog.Assert((_flags & StateFlags.PostBlockStarted) != 0, "ContextAwareResult#{0}::CaptureOrComplete|Called without calling StartPostingAsyncOp.", LoggingHash.HashString(this));
+            if ((_flags & StateFlags.PostBlockStarted) == 0 && GlobalLog.IsEnabled)
+            {
+                GlobalLog.AssertFormat("ContextAwareResult#{0}::CaptureOrComplete|Called without calling StartPostingAsyncOp.", LoggingHash.HashString(this));
+            }
 
             // See if we're going to need to capture the context.
             bool capturingContext = AsyncCallback != null || (_flags & StateFlags.CaptureContext) != 0;
 
+            bool globalLogEnabled = GlobalLog.IsEnabled;
```

Comment:
Nit: move this local above the earlier logging block and replace the usage of `GlobalLog.IsEnabled` with usage of the local.